src/cascade/input_data/db/asdr.py

```diff
@@ -111,24 +105,24 @@ def load_asdr_to_t3(execution_context) -> bool:
     """
 
     model_version_id = execution_context.parameters.model_version_id
-
     database = execution_context.parameters.database
-
-    if _asdr_in_t3(execution_context):
-        CODELOG.info(
-            f"""asdr data for model_version_id {model_version_id}
-            on '{database}' already exists, doing nothing."""
-        )
-        return False
-    else:
+    location_and_children = get_descendants(execution_context, children_only=True, include_parent=True)
+    locations_with_asdr_in_t3 = _asdr_in_t3(execution_context)
+    missing_from_t3 = set(location_and_children) - set(locations_with_asdr_in_t3)
+    if missing_from_t3:
         CODELOG.info(
             f"""Uploading asdr data for model_version_id
             {model_version_id} on '{database}'"""
         )
-
-        asdr_data = get_asdr_data(execution_context)
+        asdr_data = get_asdr_data(execution_context, list(missing_from_t3))
```

Comment:
Just because the underlying API barfs on sets, our API should accept any iterable